config/cmake/modules/FindCaliper.cmake

```diff
@@ -16,7 +16,28 @@
 
 include(MfemCmakeUtilities)
 mfem_find_package(Caliper CALIPER CALIPER_DIR
-       	"include" "caliper/cali.h"
-       	"lib" "caliper"
-  	"Paths to headers required by Caliper."
-       	"Libraries required by Caliper.")
+      "include" "caliper/cali.h"
+      "lib" "caliper"
+      "Paths to headers required by Caliper."
+      "Libraries required by Caliper.")
+
+# Append adiak path/lib if the user provided ADIAK_DIR
+if(ADIAK_DIR AND EXISTS ${ADIAK_DIR})
+    list(APPEND CALIPER_INCLUDE_DIRS ${ADIAK_DIR}/include)
+    
+    unset(_lib)
+    find_library(_lib "adiak" NO_CACHE PATHS ${ADIAK_DIR}/lib)
+    list(APPEND CALIPER_LIBRARIES ${_lib})
+endif()
+
+# Append gotcha path/lib if the user provided GOTCHA_DIR
+if(GOTCHA_DIR AND EXISTS ${GOTCHA_DIR})
+    list(APPEND CALIPER_INCLUDE_DIRS ${GOTCHA_DIR}/include)
+    
+    unset(_lib)
+    find_library(_lib "gotcha" NO_CACHE PATHS ${GOTCHA_DIR}/lib)
+    list(APPEND CALIPER_LIBRARIES ${_lib})
```

Comment:
I couldn't test GOTCHA on my Mac (works on Linux only?) but I think this can be modified similar to my suggestions about Adiak.

Reply:
Thanks -- I was able to use `find_package` here too.